troubleshooting.md

```diff
@@ -0,0 +1,211 @@
+kubectl get -A clusters
+```
+For machines run:
+```bash
+kubectl get -A machines
+```
+
+To see the details of specific resources use:
+
+For clusters:
+```bash
+kubectl describe cluster -n docker-cluster-namespace docker-cluster-one #use the namespace and name of the desired cluster
+```
+For machines:
+```bash
+kubectl describe machine -n docker-cluster-namespace docker-cluster-one-md-fjkd-309127 #use the name of the desired machine
+```
+
+The list of conditions on objects, and the messages associated with them, may help point to which controller is responsible for the issue.
+
+## 4) Check the underlying infrastructure
+This tutorial uses Cluster API Provider Docker for infrastructure - that means that each node is running in a Docker container.
+
+To see these containers run:
+```bash
+docker ps
+```
+With a single cluster `docker-cluster-one` with one control plane node and one worker node you should see output like the following:
+
+```bash
+0c7669d2b090   kindest/node:v1.24.6                 "/usr/local/bin/entr…"   30 seconds ago       Up 29 seconds                                              docker-cluster-one-md-0-x2qmh-5596d8b774-72nsr
+69c412fd86d9   kindest/node:v1.24.6                 "/usr/local/bin/entr…"   About a minute ago   Up About a minute   45305/tcp, 127.0.0.1:45305->6443/tcp   docker-cluster-one-w2vpl-7bldj
+48444f6c42f5   kindest/haproxy:v20210715-a6da3463   "haproxy -sf 7 -W -d…"   About a minute ago   Up About a minute   33877/tcp, 0.0.0.0:33877->6443/tcp     docker-cluster-one-lb
+fb9dcfb50740   kindest/node:v1.25.2                 "/usr/local/bin/entr…"   4 minutes ago        Up 4 minutes        127.0.0.1:33369->6443/tcp              kubecon-na-22-capi-lab-control-plane
+```
+The first container in the list is the worker node. Its name - at the far right column - is `docker-cluster-one-md-0` followed by a suffix. `md-0` is the name of the Machine Deployment the worker is a part of.
+The second container is the control plane node.
+The third container is a load balancer used for the control plane.
+
+## 5) Checking Cluster API component logs
+
+You may be able to find useful information about your error in the logs of the Cluster API components. Searching error strings on this troubleshooting guide, or in [the Cluster API book](https://cluster-api.sigs.k8s.io/) may point to a solution for your problem.
+For the CAPI (Core) controller manager:
+```bash
+ kubectl logs -n capd-system deployments/capd-controller-manager
+```
+For the CAPD (Docker infrastructure) controller manager:
+```bash
+ kubectl logs -n capi-system deployments/capi-controller-manager
+```
+
+For the CAPBK (Kubeadm bootstrap) controller manager:
+```bash
+ kubectl logs -n capi-kubeadm-bootstrap-system deployments/capi-kubeadm-bootstrap-controller-manager
+```
+
+For the CAPI Kubeadm control plane controller manager:
+```bash
+kubectl logs -n capi-kubeadm-control-plane-system deployments/capi-kubeadm-control-plane-controller-manager```
+````
```

Comment:
Trailing closing code block quotes + not sure if the next line needs 3 instead of 4